relay-server/src/actors/processor.rs

```diff
@@ -1449,21 +1450,29 @@ impl EnvelopeProcessorService {
         enable_cardinality_limiter: bool,
         _organization_id: u64,
         buckets: Vec<Bucket>,
-    ) -> Result<Box<dyn Iterator<Item = Bucket>>, relay_cardinality::Error> {
+    ) -> impl Iterator<Item = Bucket> {
         if !enable_cardinality_limiter {
-            return Ok(Box::new(buckets.into_iter()));
+            return Either::Left(buckets.into_iter());
```

Comment:
IIUC `Left` is for buckets that are accepted and `Right` for those that are rate-limited. Is this correct? Is it worth documenting it?

Reply:
Exactly, but it doesn't have to be like that, `Left` and `Right` can contain [any iterator](https://docs.rs/itertools/0.12.0/itertools/enum.Either.html), it's just a helper to be able to return 2 distinct iterator types as one concrete type